speaker/main.go

```diff
@@ -508,19 +516,21 @@ func (c *controller) SetConfig(l log.Logger, cfg *config.Config) controllers.Syn
 }
 
 func (c *controller) SetNode(l log.Logger, node *v1.Node) controllers.SyncState {
+	c.nodes[node.Name] = node
+
 	for proto, handler := range c.protocolHandlers {
 		if err := handler.SetNode(l, node); err != nil {
 			level.Error(l).Log("op", "setNode", "error", err, "protocol", proto, "msg", "failed to propagate node info to protocol handler")
 			return controllers.SyncStateError
 		}
 	}
-	return controllers.SyncStateSuccess
+	return controllers.SyncStateReprocessAll
```

Comment:
I know we already enforce this in the predicate, but let's be on the safe side and trigger reload only if the network status changed